juniper_official/Solutions/Pfe-wedge-detection/eachip-pattern-set.rule

```diff
@@ -0,0 +1,489 @@
+            }
+            pattern EACHIP_EVENT021 {
+                event-id EACHIP_EVENT021;
+                filter "fpc%{NUMBER:fpc_number}.*EA.*HMCIO TX: AFIFO overflow event detected in Channel";
+                description "The \"HMCIO TX: AFIFO overflow event\" messages is caused due to transient FIFO overflow condition. This indicates that the Tx AFIFO before the serdes has detected an overflow condition which is a fatal error condition from which it cannot recover without a reset.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "Contact JTAC immediately";
+                }
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT022 {
+                event-id EACHIP_EVENT022;
+                filter "fpc%{NUMBER:fpc_number}.*MQSS.*Cell packing interface error";
+                description "The \"Cell packing interface error\" messages are caused by corrupted cell received from the HMC (Hybrid Memory Cube) interfaces.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "Reset the FPC.  If on reboot, the FPC fails POST, contact JTAC for RMA.";
+                }
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT023 {
+                event-id EACHIP_EVENT023;
+                filter "fpc%{NUMBER:fpc_number}.*EA.*Checksum error detected on";
+                description "The \"Checksum error detected on\" messages are caused by corrupted cell received from the HMC (Hybrid Memory Cube) interfaces failing ECC Checksum. ECC validation has detected an error once receiving the cells from the HMC(Hybrid Memory Cube) interface.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
+                    value "Restart the FPC.  If POST errors are seen on restart, contact JTAC for RMA.  Else, monitor for recurrence.";
+                }
+                key-fields fpc_number;
+            }
+            pattern EACHIP_EVENT024 {
+                event-id EACHIP_EVENT024;
+                filter "fpc%{NUMBER:fpc_number}.*EA.*HMCIF Rx: Link.*A response packet with a FATAL state is received from HMC";
+                description "The \"response packet with a FATAL state is received from HMC with State 0x1f\" message reports a multi-bit uncorrectable error condition in the Hybrid Memory Cube. This indicates that the internal logic in the Hybrid Memory Cube has encountered a multi-bit uncorrectable error condition from which it cannot recover without a reset.";
+                constant syslog-trigger-color {
+                    value red;
+                }
+                constant syslog-trigger-message {
```

Comment:
Can we add a count function to track reccurrence?

Reply:
@ConstrainedRandomVar , added trigger based on count